Automattic/woocommerce-payments
# Add SEPA payment method.

Implements #2645 

#### Changes proposed in this Pull Request
- Added SEPA Direct Debit to UPE payment methods.
- Updated tests accordingly.
- Refactored some code to reduce redundancy.









#### Testing instructions
* Switch to this branch and `add/sepa-payment-method-fees` on server (Fee for CA is yet to be added).
* If EUR is your default currency, navigate to WooCommerce > Settings > General and change the currency to USD.
* If EUR is an enabled currency in your store, navigate to WooCommerce > Settings > Multi-Currency and remove EUR as a currency.
* Navigate to Payments > Settings.
* Click _Add payment method_, verify `SEPA Direct Debit` is in the modal, along with its icon.
![Screen Shot 2021-09-11 at 1 11 25 AM](https://user-images.githubusercontent.com/15019298/132908597-90b70f50-2521-40a2-9544-48d0080e4df8.png)

* Select `SEPA Direct Debit`, a notice should appear stating that Euros are being added to your store. Click _Add selected_, then scroll to the bottom and click _Save changes_.
![image](https://user-images.githubusercontent.com/15019298/132905868-49ff5e60-c653-4037-9324-8ed3991e7a81.png)

* Verify `SEPA Direct Debit` and its logo are now in the _Payment methods_ area.
![image](https://user-images.githubusercontent.com/15019298/132906039-1e3c30d8-b4ad-4ebf-9643-746fcf060427.png)

* Navigate to WooCommerce > Settings > Multi-Currency and verify EUR is now an enabled currency in your store.
* On the frontend, add an item to your cart, and go to checkout.
* Verify your currency *is not* EUR, payment method should show as _Credit card / debit card_ and when selecting _Use a new payment method_, `SEPA Direct Debit` should not appear.
* Switch your currency to EUR from the top right corner drop-down, the payment method should show as _Popular payment methods_ and when selecting _Use a new payment method_, `SEPA Debit` should appear.

![image](https://user-images.githubusercontent.com/15019298/132906594-43f844a0-9c5d-448c-bd0f-82cc6cb46d66.png)
* Select `SEPA Debit`.
![Screen Shot 2021-09-11 at 1 13 09 AM](https://user-images.githubusercontent.com/15019298/132908889-49e157dc-09d2-4955-b9a0-cacec5620b4d.png)

* Complete checkout/place order using IBANs given [here](https://stripe.com/docs/payments/sepa-debit/accept-a-payment?platform=checkout#test-integration).
* Once an order is received, navigate to WooCommerce > Orders in the admin.
* Find your order and select it.
* Click _Refund_, and then increase a line item in your order to 1.
* Click _Refund via WooCommerce Payments_, verify your refund completed.
* Navigate to Payments > Transactions, you should see your _Payment_ and _Payment refund_ listed for your order, verify both have the SEPA logo.

![Screen Shot 2021-09-11 at 1 08 36 AM](https://user-images.githubusercontent.com/15019298/132908318-659d272f-7123-490b-8054-65c79393b744.png)

* Click on the Payment, then verify the _Payment method_ both at the top and bottom are correctly formatted.
![Screen Shot 2021-09-11 at 1 15 55 AM](https://user-images.githubusercontent.com/15019298/132909156-12c80700-d60d-4d54-a124-502bc037f32d.png)
![image](https://user-images.githubusercontent.com/15019298/132909271-309fa60a-06a9-4297-9621-7b739dffaef3.png)

* Navigate to Payments > Settings, then click the trash can icon to delete SEPA Debit. Verify the name and the logo in the modal are correct, then click _Remove_, scroll to the bottom and _Save changes_.
* VerifySEPA debit is no longer in the list of _Payment methods_.
![image](https://user-images.githubusercontent.com/15019298/132909428-cfe3ab9d-3dcb-4991-a433-b9ca29134597.png)

-------------------
